subsys/bluetooth/host/hci_core.c

```diff
@@ -967,6 +965,33 @@ static int hci_le_read_remote_features(struct bt_conn *conn)
 	return 0;
 }
 
+int hci_read_remote_version(struct bt_conn *conn)
```

Comment:
Should this be enclosed in `#ifdef CONFIG_BT_REMOTE_VERSION` ?

Reply:
It should be static atleast.

I think using IS_ENABLED(CONFIG_BT_REMOTE) { hci_read_remote_version() } is enough to optimize it out. So it should not be included this way.
Checking with `nm` there is no hci_read_remote_version in the zephyr.elf file.